### ISSUE TITLE
Setting the `selectableTimes` causes some times to be unselectable 

### ISSUE THREAD

**Issue by jacobminer** [wdullaer/MaterialDateTimePicker#287]
My selectable times:
[9:00, 9:30, **10:30**, 11:00, 11:30, 12:00, 12:30, 13:00. 13:30, 14:00, 14:30, 15:00, 15:30, 16:00, 16:30]

The library successfully shows 10am as an hourly option (as it should, so we can get to 10:30am), but selecting 10am as the hour when the preview reads xx:00am fails to go to 10:30am. Instead, 9:30 is selected. 

If the preview reads xx:30am, I can select 10am and it will lead to 10:30.

Because this is kind of hard to explain, I've attached a video.

[screencapture-1481911902506.mp4.zip](https://github.com/wdullaer/MaterialDateTimePicker/files/657926/screencapture-1481911902506.mp4.zip)

Basically, if I try to select hour 10, and the minutes is not already at 30, the library selects hour 9 and sets the minutes to 30, instead of selecting hour 10 and minutes 30.

**Comment by wdullaer** [comment]
This is due to how the rounding is down.
Changing the hour to 10 essentially puts you in an unacceptable time. The system will then round the selection to the nearest acceptable one. Both 10:30 and 9:30 are an equal distance away, so it will round down as a tie breaker.

I should probably improve this so it first tries to round down without changing the hour.

**Comment by jacobminer** [comment]
@wdullaer I can still reproduce this on 3.1.1, although I couldn't on 3.1.0.
[screencapture-1485796448231.mp4.zip](https://github.com/wdullaer/MaterialDateTimePicker/files/739687/screencapture-1485796448231.mp4.zip)

<img width="480" alt="screen shot 2017-01-30 at 9 15 08 am" src="https://cloud.githubusercontent.com/assets/13003765/22433325/d8405190-e6cc-11e6-8406-50115f01a75f.png">

**Comment by wdullaer** [comment]
What's the issue in this video?
I can't see any problems with the rounding there.
You select 10 and it will round to 10h30, rather than 9h30 previously.

**Comment by jacobminer** [comment]
I cannot select 10:30 when my current selection is 11:00, despite 10:30 being a selectable candidate. I can only select 10:30 when my current selection is 9:30 or 11:30. Same issue as before.

**Comment by wdullaer** [comment]
Try the current master branch

**Comment by jacobminer** [comment]
Now it looks like if I attempt to select 10:30 with 9:00 or 11:00 selected, it auto-selects 9:30, instead of just selecting 10:30. 

The current functionality is
> 9:00, 9:30, 10:30, 11:00, and 11:30 are selectable times
If I have 30 minutes selected, I can select any available hour
If I have 00 minutes selected, when I touch 10:00, it will auto select 9:30, but otherwise I can select the available hours.

The functionality I'm looking for is this:
> 9:00, 9:30, 10:30, 11:00, and 11:30 are selectable times
I can touch 9, 10, or 11, regardless of what the minutes are set to
The minutes will update to the first available time upon choosing an hour

[screencapture-1485911917380.mp4.zip](https://github.com/wdullaer/MaterialDateTimePicker/files/743642/screencapture-1485911917380.mp4.zip)

**Comment by wdullaer** [comment]
This should be fixed in the latest master branch, please try that one

**Comment by jacobminer** [comment]
Looks good! Thanks!
